pkg/apis/build/v1beta1/build_types.go

```diff
@@ -0,0 +1,270 @@
+	// UndefinedVolume indicates that volume defined by build is not found in the strategy
+	UndefinedVolume BuildReason = "UndefinedVolume"
+	// TriggerNameCanNotBeBlank indicates the trigger condition does not have a name
+	TriggerNameCanNotBeBlank BuildReason = "TriggerNameCanNotBeBlank"
+	// TriggerInvalidType indicates the trigger type is invalid
+	TriggerInvalidType BuildReason = "TriggerInvalidType"
+	// TriggerInvalidGitHubWebHook indicates the trigger type GitHub is invalid
+	TriggerInvalidGitHubWebHook BuildReason = "TriggerInvalidGitHubWebHook"
+	// TriggerInvalidImage indicates the trigger type Image is invalid
+	TriggerInvalidImage BuildReason = "TriggerInvalidImage"
+	// TriggerInvalidPipeline indicates the trigger type Pipeline is invalid
+	TriggerInvalidPipeline BuildReason = "TriggerInvalidPipeline"
+
+	// AllValidationsSucceeded indicates a Build was successfully validated
+	AllValidationsSucceeded = "all validations succeeded"
+)
+
+// BuildReasonPtr returns a pointer to the passed BuildReason.
+func BuildReasonPtr(s BuildReason) *BuildReason {
+	return &s
+}
+
+// ConditionStatusPtr returns a pointer to the passed ConditionStatus.
+func ConditionStatusPtr(s corev1.ConditionStatus) *corev1.ConditionStatus {
+	return &s
+}
+
+const (
+	// BuildDomain is the domain used for all labels and annotations for this resource
+	BuildDomain = "build.shipwright.io"
+
+	// LabelBuild is a label key for defining the build name
+	LabelBuild = BuildDomain + "/name"
+
+	// LabelBuildGeneration is a label key for defining the build generation
+	LabelBuildGeneration = BuildDomain + "/generation"
+
+	// AnnotationBuildRefSecret is an annotation that tells the Build Controller to reconcile on
+	// events of the secret only if is referenced by a Build in the same namespace
+	AnnotationBuildRefSecret = BuildDomain + "/referenced.secret"
+
+	// AnnotationBuildVerifyRepository tells the Build Controller to check a remote repository. If the annotation is not set
+	// or has a value of 'true', the controller triggers the validation. A value of 'false' means the controller
+	// will bypass checking the remote repository.
+	AnnotationBuildVerifyRepository = BuildDomain + "/verify.repository"
+)
+
+// BuildSpec defines the desired state of Build
+type BuildSpec struct {
+	// Source refers to the Git repository containing the
+	// source code to be built.
+	Source Source `json:"source"`
+
+	// Trigger defines the scenarios where a new build should be triggered.
+	//
+	// +optional
+	Trigger *Trigger `json:"trigger,omitempty"`
+
+	// Strategy references the BuildStrategy to use to build the container
+	// image.
+	Strategy Strategy `json:"strategy"`
+
+	// Params is a list of key/value that could be used
+	// to set strategy parameters
+	//
+	// +optional
+	ParamValues []ParamValue `json:"paramValues,omitempty"`
+
+	// Output refers to the location where the built image would be pushed.
+	Output Image `json:"output"`
```

Comment:
Imo the type should be named Output like the key, that would be consistent with what we do anywhere else.

Reply:
I agree, however I prefer this to be a separated item via its own issue.